File: TestResults/linter/src/personViewModel.ts
import {Gender, ViewModel} from "./models.latest";

/**
 * The PersonViewModel that represents a Person.
 * @extends ViewModel
 */
export class PersonViewModel extends ViewModel {
    /** The Identifier that this Model refers to. */
    identifier: string = "e5fef186-8cd4-44e7-b60a-091a8e916683";
    /** The Name of the person. */
    name!: string;
    /** The Surname of the person. */
    surname!: string;
    /** The Gender of the person. */
    gender: Gender | number = 1;

    constructor(value: any = null) {
        super(value);

        if (Object.prototype.hasOwnProperty.call(value, 'identifier'))
            this.identifier = value.identifier;

        if (Object.prototype.hasOwnProperty.call(value, 'name'))
            this.name = value.name;

        if (Object.prototype.hasOwnProperty.call(value, 'surname'))
            this.surname = value.surname;

        if (Object.prototype.hasOwnProperty.call(value, 'gender'))
            this.gender = value.gender;

        if (Object.prototype.hasOwnProperty.call(value, 'guid'))
            this.guid = value.guid;

        if (Object.prototype.hasOwnProperty.call(value, 'createdOn'))
            this.createdOn = value.createdOn;

    }
}

File: Sushi.TestModels/PersonViewModel.cs
namespace Sushi.TestModels;

/// <summary>
///		The <see cref="PersonViewModel"/> that represents a Person.
/// </summary>
public class PersonViewModel : ViewModel
{
    /// <summary>
    ///     The <see cref="Identifier"/> that this Model refers to.
    /// </summary>
    public Guid Identifier { get; set; } = Guid.NewGuid();

    /// <summary>
    ///		 The <see cref="Name"/> of the person.
    /// </summary>
    public string Name { get; set; } = "Jeroen";

    /// <summary>
    ///		The <see cref="Surname"/> of the person.
    /// </summary>
    public string Surname { get; set; } = "Vorsselman";

    /// <summary>
    ///		The <see cref="Gender"/> of the person.
    /// </summary>
    public Gender Gender { get; set; } = Gender.Male;

    public PersonViewModel(string name, string surname)
    {
        Name = name;
        Surname = surname;
    }
}